pywr/core.py

```diff
@@ -118,9 +132,22 @@ def __init__(self, solver=None, parameters=None):
 
         self.node = {}
         self.group = {}
+        self.scenarios = ScenarioCollection()
 
         self.reset()
 
+    @property
+    def number_of_timesteps(self):
+        return len(self.timestepper)
+
+    @property
+    def number_of_scenarios(self):
```

Comment:
Why can't we use `itertools.product` for this anyway?

https://docs.python.org/2/library/itertools.html#itertools.product

```
>>> [x for x in itertools.product([1,2],[3,4],[5,6])]
[(1, 3, 5), (1, 3, 6), (1, 4, 5), (1, 4, 6), (2, 3, 5), (2, 3, 6), (2, 4, 5), (2, 4, 6)]
```